packages/datadog-plugin-aws-sdk/src/services/kinesis.js

```diff
@@ -1,15 +1,64 @@
 'use strict'
-
+const log = require('../../../dd-trace/src/log')
 class Kinesis {
   generateTags (params, operation, response) {
-    const tags = {}
-
-    if (!params || !params.StreamName) return tags
+    if (!params || !params.StreamName) return {}
 
-    return Object.assign(tags, {
+    return {
       'resource.name': `${operation} ${params.StreamName}`,
       'aws.kinesis.stream_name': params.StreamName
-    })
+    }
+  }
+
+  // AWS-SDK will b64 kinesis payloads
+  // or will accept an already b64 encoded payload
+  // This method handles both
+  _tryParse (body) {
+    try {
+      return JSON.parse(body)
+    } catch (e) {
```

Comment:
Creating an error is quite heavy in JavaScript. Any way to validate if it's JSON with a quick look at the beginning of the string?

Reply:
We could check if the string starts with `'{"`, but that's not really a guarantee and could throw errors later as we try to inject trace context. After talking with the Python team, we thought this was the best option.

Reply:
It's possible that in Python creating an error is lightweight, but in JavaScript it's disastrous for performance, so it should only be used when absolutely necessary, especially for instrumentation purpose. If the format is standard and always start with `{` that would definitely be a good start. I guess the rule will depend on the format expected by AWS.

Is there any way we can get the data before it's been serialized? At the end of the day if we have no other choice we'll have to do it the current way but generally speaking we should try to avoid having to parse/re-serialize, and _really_ try to avoid creating errors when possible.

Reply:
> It's possible that in Python creating an error is lightweight, but in JavaScript it's disastrous for performance, so it should only be used when absolutely necessary, especially for instrumentation purpose. If the format is standard and always start with { that would definitely be a good start. I guess the rule will depend on the format expected by AWS.
Okay, so your suggestion is to check if the string starts with `'{`, and if not, try the base64 route first?


> Is there any way we can get the data before it's been serialized? At the end of the day if we have no other choice we'll have to do it the current way but generally speaking we should try to avoid having to parse/re-serialize, and really try to avoid creating errors when possible.

Not really, most of the time users serialize this as a json string themselves before making the API call.

Reply:
> Okay, so your suggestion is to check if the string starts with '{, and if not, try the base64 route first?

It depends as I don't know what format AWS expects. If it has to be an object, then checking for `{` would be sufficient.

> Not really, most of the time users serialize this as a json string themselves before making the API call.

Ok, then there is unfortunately no way around it.